[Bug] [CLI] Failed to apply Kyverno policies `validate.CEL` subrules using CLI

### Problem Statement

1. Create a policy that uses `validate.cel` subrule as follows:
```
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: check-deployment-replicas
spec:
  validationFailureAction: Enforce
  background: false
  rules:
    - name: deployment-replicas
      match:
        any:
        - resources:
            kinds:
              - Deployment
      validate:
        cel:
          expressions:
            - expression: "object.spec.replicas <= 3"
              message: "Deployment spec.replicas must be less than 3."
```
2. Create a deployment:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: deployment-test
spec:
  replicas: 1
  selector:
    matchLabels:
      app: app
  template:
    metadata:
      labels:
        app: app
    spec:
      containers:
      - name: container2
        image: nginx
```
3. Apply the policy on the resource using CLI without `--cluster`:
```
./cmd/cli/kubectl-kyverno/kubectl-kyverno apply ../vaps/check-deployment-replicas.yaml --resource ../vaps/deployment.yaml 

Applying 1 policy rule(s) to 1 resource(s)...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x2751adf]

goroutine 1 [running]:
github.com/kyverno/kyverno/pkg/engine/adapters.(*dclientAdapter).GetNamespace(0xc004163b05?, {0x4172438, 0xc000076168}, {0xc004163b60, 0xa}, {{{0x0, 0x0}, {0x0, 0x0}}, {0x0, ...}})
        /home/mariamfahmy/Projects/Kyverno/kyverno/pkg/engine/adapters/dclient.go:50 +0x3f
github.com/kyverno/kyverno/pkg/engine/handlers/validation.validateCELHandler.Process({{_, _}}, {_, _}, {{_, _}, _}, {_, _}, {0xc00418e960}, ...)
        /home/mariamfahmy/Projects/Kyverno/kyverno/pkg/engine/handlers/validation/validate_cel.go:149 +0x1737
github.com/kyverno/kyverno/pkg/engine.(*engine).invokeRuleHandler.func1({0x4172438, 0xc000076168}, {0x3710938?, 0xa?})
        /home/mariamfahmy/Projects/Kyverno/kyverno/pkg/engine/engine.go:292 +0x82e
github.com/kyverno/kyverno/pkg/tracing.ChildSpan2[...]({0x4172438?, 0xc000076168?}, {0x3710938?, 0xc000680400?}, {0xc004067c80?, 0xc0041f6740?}, 0xc0044805e8?, {0x0, 0x0, 0x0})
        /home/mariamfahmy/Projects/Kyverno/kyverno/pkg/tracing/childspan.go:59 +0xf3
github.com/kyverno/kyverno/pkg/engine.(*engine).invokeRuleHandler(_, {_, _}, {{_, _}, _}, _, {_, _}, {0xc00418e960}, ...)
        /home/mariamfahmy/Projects/Kyverno/kyverno/pkg/engine/engine.go:244 +0x1ed
github.com/kyverno/kyverno/pkg/engine.(*engine).validate(0xc001016210, {0x4172438, 0xc000076168}, {{0x417dda8?, 0xc0041f6640?}, 0x4189ca0?}, {0x4190920, 0xc0040c32c0})
        /home/mariamfahmy/Projects/Kyverno/kyverno/pkg/engine/validation.go:64 +0x3f7
github.com/kyverno/kyverno/pkg/engine.(*engine).Validate(0xc001016210, {0x4172438, 0xc000076168}, {0x4190920, 0xc0040c32c0})
        /home/mariamfahmy/Projects/Kyverno/kyverno/pkg/engine/engine.go:95 +0x185
github.com/kyverno/kyverno/cmd/cli/kubectl-kyverno/utils/common.ApplyPolicyOnResource({{0x41acf48, 0xc000c20a00}, {{{0x0, 0x0}, {0x0, 0x0}}, {{0x0, 0x0}, {0x0, 0x0}, ...}, ...}, ...})
        /home/mariamfahmy/Projects/Kyverno/kyverno/cmd/cli/kubectl-kyverno/utils/common/kyverno_policies_types.go:179 +0x1557
github.com/kyverno/kyverno/cmd/cli/kubectl-kyverno/apply.(*ApplyCommandConfig).applyPolicytoResource(_, _, {_, _, _}, {_, _, _}, {0xc004116020, 0x1, ...}, ...)
        /home/mariamfahmy/Projects/Kyverno/kyverno/cmd/cli/kubectl-kyverno/apply/apply_command.go:356 +0xb18
github.com/kyverno/kyverno/cmd/cli/kubectl-kyverno/apply.(*ApplyCommandConfig).applyCommandHelper(0xc00100c1a0)
        /home/mariamfahmy/Projects/Kyverno/kyverno/cmd/cli/kubectl-kyverno/apply/apply_command.go:250 +0x865
github.com/kyverno/kyverno/cmd/cli/kubectl-kyverno/apply.Command.func1(0xc000fff500?, {0xc000fb72c0, 0x1, 0x3})
        /home/mariamfahmy/Projects/Kyverno/kyverno/cmd/cli/kubectl-kyverno/apply/apply_command.go:170 +0x130
github.com/spf13/cobra.(*Command).execute(0xc000fff500, {0xc000fb7260, 0x3, 0x3})
        /home/mariamfahmy/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc000ffe000)
        /home/mariamfahmy/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/mariamfahmy/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
main.main()
        /home/mariamfahmy/Projects/Kyverno/kyverno/cmd/cli/kubectl-kyverno/main.go:28 +0x7b
```
4. Apply the policy again using CLI with `--cluster`:
```
./cmd/cli/kubectl-kyverno/kubectl-kyverno apply ../vaps/check-deployment-replicas.yaml --cluster --policy-report

Applying 1 policy rule(s) to 3 resource(s)...
----------------------------------------------------------------------
POLICY REPORT:
----------------------------------------------------------------------
apiVersion: wgpolicyk8s.io/v1alpha2
kind: ClusterPolicyReport
metadata:
  creationTimestamp: null
  name: clusterpolicyreport
results:
- message: Validation rule 'deployment-replicas' passed.;
  policy: check-deployment-replicas
  resources:
  - apiVersion: apps/v1
    kind: Deployment
    name: deployment-test
    namespace: default
    uid: 48e4fcc5-fba1-481e-82d0-a3362782f53e
  result: pass
  rule: deployment-replicas
  scored: true
  source: kyverno
  timestamp:
    nanos: 0
    seconds: 1692792537
- message: Validation rule 'deployment-replicas' passed.;
  policy: check-deployment-replicas
  resources:
  - apiVersion: apps/v1
    kind: Deployment
    name: coredns
    namespace: kube-system
    uid: f5ca778d-60ec-4710-8e12-3099212f61c7
  result: pass
  rule: deployment-replicas
  scored: true
  source: kyverno
  timestamp:
    nanos: 0
    seconds: 1692792537
- message: Validation rule 'deployment-replicas' passed.;
  policy: check-deployment-replicas
  resources:
  - apiVersion: apps/v1
    kind: Deployment
    name: local-path-provisioner
    namespace: local-path-storage
    uid: 65595044-e731-4438-84ed-fae5decb287f
  result: pass
  rule: deployment-replicas
  scored: true
  source: kyverno
  timestamp:
    nanos: 0
    seconds: 1692792537
summary:
  error: 0
  fail: 0
  pass: 3
  skip: 0
  warn: 0
```

Conclusion:
Applying `validate.cel` subrules on resources work as expected when `--cluster` is set. Otherwise, it panics because `NewValidateCELHandler.Process` uses the client which is null in the 1st case. 

### Solution Description

One possible solution is to check if client is null in `NewValidateCELHandler`.

### Alternatives

_No response_

### Additional Context

_No response_

### Slack discussion

_No response_

### Research

- [X] I have read and followed the documentation AND the [troubleshooting guide](https://kyverno.io/docs/troubleshooting/).
- [X] I have searched other issues in this repository and mine is not recorded.

--- a/cmd/cli/kubectl-kyverno/utils/common/kyverno_policies_types.go
+++ b/cmd/cli/kubectl-kyverno/utils/common/kyverno_policies_types.go
@@ -113,12 +113,16 @@ OuterLoop:
 			}
 		}
 	}
+	var client engineapi.Client
+	if c.Client != nil {
+		client = adapters.Client(c.Client)
+	}
 	rclient := registryclient.NewOrDie()
 	eng := engine.NewEngine(
 		cfg,
 		config.NewDefaultMetricsConfiguration(),
 		jmespath.New(cfg),
-		adapters.Client(c.Client),
+		client,
 		factories.DefaultRegistryClientFactory(adapters.RegistryClient(rclient), nil),
 		imageverifycache.DisabledImageVerifyCache(),
 		store.ContextLoaderFactory(nil),

--- a/pkg/engine/handlers/validation/validate_cel.go
+++ b/pkg/engine/handlers/validation/validate_cel.go
@@ -77,7 +77,7 @@ func (h validateCELHandler) Process(
 	auditExpressions := convertAuditAnnotations(auditAnnotations)
 
 	// get the parameter resource if exists
-	if hasParam {
+	if hasParam && h.client != nil {
 		paramKind := rule.Validation.CEL.GetParamKind()
 		paramRef := rule.Validation.CEL.GetParamRef()
 
@@ -122,7 +122,7 @@ func (h validateCELHandler) Process(
 	if gvk.Kind == "Namespace" && gvk.Version == "v1" && gvk.Group == "" {
 		namespaceName = ""
 	}
-	if namespaceName != "" {
+	if namespaceName != "" && h.client != nil {
 		namespace, err = h.client.GetNamespace(ctx, namespaceName, metav1.GetOptions{})
 		if err != nil {
 			return resource, handlers.WithResponses(

